File: c++/07_pgm_thread/src/co/edu/campusucc/poo/ThreadRunnable.cpp
#include <iostream>
#include <thread>

using namespace std;

class ThreadRunnable : public Runnable {
private:
int times;

public:
void run() {
cout << "Runnable Thread Started Running...on UCC University\n";
cout << "Random Times: " << times << endl;
try {
printTimes();
} catch (const exception& e) {
cerr << e.what() << endl;
}
}
void printTimes() {
    int timesCount = 0;
    while (timesCount <= times) {
        cout << "-";
        timesCount++;
        this_thread::sleep_for(chrono::milliseconds(50));
    }
    cout << "> OK\n";
}

int getTimes() {
    return times;
}

void setTimes(int times) {
    this->times = times;
}
};

File: python/java_to_python/07-pgmThread/src/co/edu/campusucc/poo/domain/ThreadRunnable.java
package co.edu.campusucc.poo.domain;

public class ThreadRunnable implements Runnable{
    //Properties
    private int times;

    public void run()
    {
        System.out.println("Runnable Thread Started Running...on UCC University");
        System.out.println("Random Times: "+times);
        try {
            printTimes();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    //Defined Behavior
    public void printTimes() throws Exception{
        int timesCount = 0;
        while (timesCount<=times) {
            System.out.print("-");
            timesCount++;
            Thread.sleep(50);
        }
        System.out.println("> OK");
    }

    //Getters and Setters
    public int getTimes() {
        return times;
    }

    public void setTimes(int times) {
        this.times = times;
    }
}
